lua/neoclip/telescope.lua

```diff
@@ -162,6 +162,8 @@ local function get_export(register_names, typ)
         if opts ~= nil and opts.extra ~= nil then
             register_names = utils.join(register_names, parse_extra(opts.extra))
         end
+        -- NOTE: use system clipboard
+        handlers.handle_system_yank()
```

Comment:
wont this only show what's currently in eg `"+` but not the history as for other registers?

Reply:
we would need some kind of event/trigger whenever these registers are changed from system but not sure that's possible, any ideas?

Reply:
no, history in other registers still can be seen

it just add "* and "+ into storage.yank additionally

Reply:
indeed, but i dont find any event related to "register changed". 
maybe, we can simply set a command right now.

Reply:
you mean a command/function that a user can call to add entries to the history? If so, would calling
```lua
require('neoclip.storage').({
    regtype = regtype,
    contents = regcontents,
    filetype = vim.bo.filetype,
}, 'yanks')
```
work?

Reply:
define a user event
```vim
autocmd User RegStarChange lua require("neoclip.handlers").handle_system_yank()
```

trigger the event
```
doautocmd User RegStarChange
```

But still can not be auto triggered, still weird

Reply:
take a look at this [issue](https://github.com/neovim/neovim/issues/15136)

Reply:
What would `require("neoclip.handlers").handle_system_yank()` do differently than:
```lua
require('neoclip.storage').insert({
    regtype = regtype,
    contents = regcontents,
    filetype = vim.bo.filetype,
}, 'yanks')
```
? Or would that be it?

Just to make sure I understand you suggestion correctly :)

Also are you suggesting that `RegStarChange` be part of `neoclip`? I think this can be done by the user?

Reply:
> What would `require("neoclip.handlers").handle_system_yank()` do differently than:
> 
> ```lua
> require('neoclip.storage').insert({
>     regtype = regtype,
>     contents = regcontents,
>     filetype = vim.bo.filetype,
> }, 'yanks')
> ```
nothing special but make sure that it's a unified code style, just like [this](https://github.com/AckslD/nvim-neoclip.lua/blob/09fa54a2e9866ba05d08980e27b29099d6d0ed6e/lua/neoclip.lua#L11)

> Also are you suggesting that `RegStarChange` be part of `neoclip`? I think this can be done by the user?
not necessary, user can do it in their own vimrc

I've noticed that `TextPutPre` and `TextPutPost` still working in progress, right now there's no event to trigger `require("neoclip.handlers").handle_system_yank()`, sad

Reply:
I'm happy to add `handle_system_yank` if that's useful for you.

Although I don't think we should modify the registers on `delete` since we don't do that with other registers, rather it deletes from the history but the content of the registers remains.

Also I don't think `handle_system_yank` should be called in the picker. It should be the task of the user to call it prior by whatever event/action is available.

I'm not sure eg `TextPutPre` would help here though? Isn't that events triggered when pasting. But you want some trigger/event when a register is changed or when you copy something in your system clipboard?